### PR TITLE
truffle: add migrations & deployment scripts

### PR DESCRIPTION
**Pull request by crisog**
Adding a deploy script for TokenGeyser contract.

To deploy the contracts, just:
1. Configure your `.env` file.
2. Update the constants with RPC URLs & desired gas prices.
3. Run `npx truffle migrate --reset --network <network_name>`

### CODE REVIEW

**Comment by Evalir** [comment]
Not very well versed on using truffle (yes I know, I'm spoiled)—could we get some docs on how to use this/examples?

File: truffle.js [hunk 1]
```diff
@@ -26,8 +10,22 @@ module.exports = {
     mainnet: {
       ref: 'mainnet-prod',
       network_id: 1,
-      provider: () => keystoreProvider(mainnetUrl),
+      provider: () => {
+        return new HDWalletProvider(
+          [process.env.PRIVATE_KEY],
+          'https://eth-mainnet.gateway.pokt.network/v1/<APP_ID>');
+      },
       gasPrice: 35000000000
```
Comment (Evalir) [inline truffle.js]: Why are we setting the gas price manually for mainnet? Shouldn't we be able to figure it out at runtime? If not, I'd put both this and the rinkeby gas price as a constant above, like `MAINNET_GAS_PRICE` and `RINKEBY_GAS_PRICE`

**Review by Evalir — APPROVED** [review]
LGTM! ❤️

File: truffle.js (hunk 1 above)
Comment (crisog) [reply truffle.js]: > Why are we setting the gas price manually for mainnet? Shouldn't we be able to figure it out at runtime? If not, I'd put both this and the rinkeby gas price as a constant above, like `MAINNET_GAS_PRICE` and `RINKEBY_GAS_PRICE`

It's not common to create deployment scripts that use APIs for fetching gas prices. It's possible, but not even ideal if who deploys the contract knows what they are doing.

Comment (Evalir) [reply truffle.js]: Yeah of course, I think it makes sense with volatile gas prices right now, but Ethers does actually a pretty good job of figuring out gas prices... Even though this is using web3, so, makes sense 😆

**Pull request merged**